common/src/main/java/io/netty/util/concurrent/FastThreadLocal.java

```diff
@@ -123,10 +123,22 @@ private static void removeFromVariablesToRemove(
         variablesToRemove.remove(variable);
     }
 
+    // visible for tests
+    static boolean hasOnRemoval(Class<? extends FastThreadLocal> clazz) {
+      try {
+        Class<?> declaringClass = clazz.getMethod("onRemoval", Object.class).getDeclaringClass();
```

Comment:
I know you added some tests for this already, but should we note that the method name itself is important due to reflective usage at the `onRemoval` declaration?